### ISSUE TITLE
shorthand loops result in type::Any

### ISSUE THREAD

**Issue by ipozdeev** [JuliaLang/julia#14916]
Hi all,

when I'm trying to construct a simple autoregressive variable in a shorthand loop:
```
u = randn(10);
e = u;
e = [t == 1 ? u[t] : e[t-1]*0.8 + u[t] for t = 1:10]
```
e is suddenly of type "Array{Any,10}", though u is of type "Vector{Float64,10} which later messes up with plots. Hence a question: what causes this and how can I get a Float64 representation without resorting to a "usual" loop (which works fine, btw).

On a side note, how do I convert a Vector of "Any" type to a Vector of type "Float64"?

Thanks.

\igor

**Comment by simonster** [comment]
Type information is not available at the global scope, but you can tell a comprehension to generate a `Vector{Float64}` by using something like `Float64[x for i = 1:10]`. In general, please direct these kinds of questions to https://groups.google.com/forum/#!forum/julia-users

**Comment by JeffBezanson** [comment]
Also this will be fixed soon by #7258.
